### PR TITLE
ghostscript 9.53.1

### PR DESCRIPTION
**Pull request by seisman**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by miccal** [comment]
@seisman if it is not too much trouble, could you also change the `regex` in the `livecheck` block to:
```
regex(/^ghostpdl[._-]v?(\d+(?:\.\d+)+)$/i)
```
Would you agree with this @samford?

File: Formula/ghostscript.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class Ghostscript < Formula
   desc "Interpreter for PostScript and PDF"
   homepage "https://www.ghostscript.com/"
-  url "https://github.com/ArtifexSoftware/ghostpdl-downloads/releases/download/gs952/ghostpdl-9.52.tar.gz"
-  sha256 "8f6e48325c106ae033bbae3e55e6c0b9ee5c6b57e54f7cd24fb80a716a93b06a"
+  url "https://github.com/ArtifexSoftware/ghostpdl-downloads/releases/download/gs9530/ghostpdl-9.53.0.tar.gz"
+  sha256 "8b937ab904aa91dbbfe8016f5cab48c64e9d8269e07ad2250d0b8d76c0138637"
   license "AGPL-3.0"
```
Comment (miccal) [inline Formula/ghostscript.rb:6]: Please change this to:
```
license "AGPL-3.0-or-later"
```

**Comment by samford** [comment]
> `regex(/^ghostpdl[._-]v?(\d+(?:\.\d+)+)$/i)`

Yes, this is what the `livecheck` block `regex` should be doing, since we use `ghostpdl` releases in this formula. This will fix the issue with `livecheck` not reporting the correct latest `ghostpdl` version.

File: Formula/ghostscript.rb (hunk 1 above)
Comment (seisman) [reply Formula/ghostscript.rb]: Done.

**Comment by seisman** [comment]
> @seisman if it is not too much trouble, could you also change the `regex` in the `livecheck` block to:
> 
> ```
> regex(/^ghostpdl[._-]v?(\d+(?:\.\d+)+)$/i)
> ```
> 
Done.

**Comment by seisman** [comment]
It seems imagemagick fails. Do I need to bump the `revision` of imagemagick in this PR?
```
==> brew install --only-dependencies imagemagick
==> brew linkage --test imagemagick
==> FAILED
Broken dependencies:
  /usr/local/opt/ghostscript/lib/libgs.9.52.dylib (ghostscript)
No broken library linkage detected
```

**Comment by miccal** [comment]
> It seems imagemagick fails. Do I need to bump the `revision` of imagemagick in this PR?
> 
> ```
> ==> brew install --only-dependencies imagemagick
> ==> brew linkage --test imagemagick
> ==> FAILED
> Broken dependencies:
>   /usr/local/opt/ghostscript/lib/libgs.9.52.dylib (ghostscript)
> No broken library linkage detected
> ```

@seisman yes I believe so.

File: Formula/libspectre.rb [hunk 2]
```diff
@@ -4,6 +4,7 @@ class Libspectre < Formula
   url "https://libspectre.freedesktop.org/releases/libspectre-0.2.9.tar.gz"
   sha256 "49ae9c52b5af81b405455c19fe24089d701761da2c45d22164a99576ceedfbed"
   license "GPL-2.0"
```
Comment (miccal) [inline Formula/libspectre.rb:6]: Please change this to:
```
license "GPL-2.0-or-later"
```
From the `homepage`:

> libspectre is free software and is available to be redistributed and/or modified under the terms of the GNU General Public License as published by the Free Software Foundation; either version 2, or (at your option) any later version.

File: Formula/ghostscript.rb [hunk 3]
```diff
@@ -49,7 +49,6 @@ def install
       --disable-gtk
       --disable-fontconfig
       --without-libidn
-      --with-system-libtiff
```
Comment (miccal) [inline Formula/ghostscript.rb:52]: Can you please explain why you removed this option?

File: Formula/ghostscript.rb [hunk 4]
```diff
@@ -49,7 +49,6 @@ def install
...
       --without-x
```
Comment (miccal) [inline Formula/ghostscript.rb:52]: It looks like this is no longer a valid option and is now the default (there is an option for `--with-x `), so I think this line can be removed.

File: Formula/libspectre.rb (hunk 2 above)
Comment (seisman) [reply Formula/libspectre.rb]: Done.

File: Formula/ghostscript.rb (hunk 3 above)
Comment (seisman) [reply Formula/ghostscript.rb:52]: With this option, configure gives an error:
```
configure: error: Mixing local libtiff with shared libjpeg not supported
```

File: Formula/ghostscript.rb (hunk 4 above)
Comment (seisman) [reply Formula/ghostscript.rb]: Done.

Comment (seisman) [reply Formula/ghostscript.rb]: Just tried the recipe locally. Removing `--without-x` gives me the following error.
```
In file included from ./devices/gdevxalt.c:21:
./base/x_.h:131:12: fatal error: 'X11/Xlib.h' file not found
#  include <X11/Xlib.h>
           ^~~~~~~~~~~~
1 error generated.
make: *** [obj/gdevxalt.o] Error 1
```

File: Formula/ghostscript.rb (hunk 3 above)
Comment (miccal) [reply Formula/ghostscript.rb:52]: Thank you for the explanation, much appreciated.

File: Formula/ghostscript.rb (hunk 4 above)
Comment (miccal) [reply Formula/ghostscript.rb]: Ah, ok -- please disregard my suggestion then,  sorry about that.

Comment (seisman) [reply Formula/ghostscript.rb:52]: I just added `--without-x` back.

Comment (miccal) [reply Formula/ghostscript.rb:52]: Thank you, sorry for the trouble.

File: Formula/ghostscript.rb (hunk 3 above)
Comment (SMillerDev) [reply Formula/ghostscript.rb:52]: Can't we define libjpeg as local too?

Comment (seisman) [reply Formula/ghostscript.rb:52]: I don't see flags like `--with-system-libjpeg` or similar.

**Comment by SMillerDev** [comment]
Can you make your work 1 commit per formula, using `ghostscript 9.53.0` and `imagick: revision for ghostscript`

**Comment by seisman** [comment]
> Can you make your work 1 commit per formula, using `ghostscript 9.53.0` and `imagick: revision for ghostscript`

Done.

**Comment by seisman** [comment]
FYI, ghostscript just released a patch version 9.53.1 yesterday. I've bumped the formula to the latest version.

**Comment by seisman** [comment]
Ping @SMillerDev @miccal

**Comment by miccal** [comment]
@seisman everything looks good to me, thank you (I am only a contributor, not a maintainer).

**Review by SMillerDev — COMMENTED** [review]
Thanks @seisman  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Review by SMillerDev — DISMISSED** [review]

**Comment by seisman** [comment]
Bottle published failed. Do I need to rebase this branch?

**Comment by SMillerDev** [comment]
No, GitHub CI needs to stop breaking

**Review by SMillerDev — DISMISSED** [review]

**Review by SMillerDev — DISMISSED** [review]

**Review by dtrodrigues — DISMISSED** [review]

**Comment by jonchang** [comment]
> Error: Bottles are for imagemagick 7.0.10-29_1 but formula is version 7.0.10-30_1!

This needs a rebase to rebuild the bottles.

**Comment by seisman** [comment]
> > Error: Bottles are for imagemagick 7.0.10-29_1 but formula is version 7.0.10-30_1!
> 
> This needs a rebase to rebuild the bottles.

Done.

**Review by SMillerDev — DISMISSED** [review]

**Pull request closed without merging**